lib/OpenQA/Utils.pm

```diff
@@ -257,6 +258,41 @@ sub run_cmd_with_log {
     return run_cmd_with_log_return_error($cmd)->{status};
 }
 
+sub run_cmd_with_log_return_all {
+    my ($cmd) = @_;
+    log_info("Running cmd: " . join(' ', @$cmd));
+    try {
+        my ($stdout, $stderr);
+        my $ipc_run_succeeded = IPC::Run::run($cmd, \undef, \$stdout, \$stderr);
+        my $return_code       = $?;
+        chomp $stdout;
+        chomp $stderr;
+        if ($ipc_run_succeeded) {
+            log_debug($stdout);
+            log_info("cmd returned $return_code");
+        }
+        else {
+            log_warning($stderr);
+            log_error("cmd returned $return_code");
+        }
+        return {
```

Comment:
This will only return from the `try` block because the `{}`-braces of the `try`-block actually enclose a nested function. This only works because the function doesn't to anything after the `try-catch` anyways. However, to avoid future bugs I'd write that as `eval { … }` followed by `if (my $error = $@) { … }`.